From 1b8419462a554749ea61c77bf58ef7963b984b1f Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Tue, 18 Sep 2012 11:43:26 +0530
Subject: [PORT FROM MAIN]Seek to work good if cue-elements present in .mka container (audio-only contents)

BZ: 57757

Seek was done only for video track. So seek was failing in
audio only clips. This is fixed by seeking to audio in audio
only clips.

Change-Id: I01cbe904000ff600db2d48649e3caa35700dab09
Orig-Change-Id: If178491608ad009c4188849fc71daeab841ccbd8
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |   12 ++++--------
 1 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index 84bce9e..185b9dc 100644
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -383,16 +383,12 @@ void BlockIterator::seek(
         }
     }
 
-    // Always *search* based on the video track, but finalize based on mTrackNum
+    // Always *search* based on the video track, if video is present.
     const mkvparser::CuePoint::TrackPosition* pTP;
-    if (pTrack && pTrack->GetType() == 1) {
-        bool ret = pCues->Find(seekTimeNs, pTrack, pCP, pTP);
-        if (!ret) {
-            ALOGE("Could not find the matching cue point");
-            return;
-        }
+    if (pTrack) {
+        pCues->Find(seekTimeNs, pTrack, pCP, pTP);
     } else {
-        ALOGE("Did not locate the video track for seeking");
+        ALOGE("Did not locate the track for seeking");
         return;
     }
 
-- 
1.7.4.1

